### PR TITLE
"SetInclinedToJoin" & "GetInclinedToJoin" script call implementations

### PR DESCRIPTION
**Pull request by PierceLBrooks**
This also includes a "OnMonsterInteract" callback.

### CODE REVIEW

File: src/cpp/shared/adventure/adv.cpp [hunk 1]
```diff
@@ -313,4 +313,55 @@ void advManager::QuickInfo(int x, int y) {
...
+	int x;
+	int y;
+	if (cell->objType != (LOCATION_ARMY_CAMP | TILE_HAS_EVENT)) {
+		this->PlayerMonsterInteract_orig(cell, other, player, window, a1, a2, a3, a4, a5);
+		return;
+	}
+	if (GetMapCellXY(cell, &x, &y)) {
+		ScriptCallback("OnMonsterInteract", x, y);
+	}
+	if ((x < 0) && (y < 0) && (x >= gpGame->map.width) || (y >= gpGame->map.height)) {
```
Comment (jkoppel) [inline src/cpp/shared/adventure/adv.cpp]: Why are the bounds checks necessary?

File: src/cpp/shared/adventure/adv.cpp [hunk 2]
```diff
@@ -313,4 +313,55 @@ void advManager::QuickInfo(int x, int y) {
...
+		this->PlayerMonsterInteract_orig(cell, other, player, window, a1, a2, a3, a4, a5);
+		return;
+	}
+	int cellIdx = y * gpGame->map.height + x;
+	this->PlayerMonsterInteract_orig(&gpGame->map.tiles[cellIdx], other, player, window, a1, a2, a3, a4, a5);
```
Comment (jkoppel) [inline src/cpp/shared/adventure/adv.cpp]: Isn't the first argument just "cell"?

File: src/cpp/shared/game/game.h [hunk 3]
```diff
@@ -162,7 +162,7 @@ class game {
 	char field_660D;
 	char field_660E;
 
-  // New state
+  // New states
```
Comment (jkoppel) [inline src/cpp/shared/game/game.h]: Please revert this

File: src/cpp/shared/game/game.cpp [hunk 4]
```diff
@@ -214,11 +214,11 @@ void game::PerMonth() {
 }
 
 void game::ResetIronfistGameState() {
```
Comment (jkoppel) [inline src/cpp/shared/game/game.cpp]: Revert this as well. I recommend inspecting your git status before making any commits so you don't accidentally commit whitespace changes like this.

File: src/cpp/shared/game/game.h (hunk 3 above)
Comment (PierceLBrooks) [reply src/cpp/shared/game/game.h]: Ah, this was left behind from when I was using a boolean checker grid for the WillJoin implementation. That array was declared here in a similar fashion, so it made sense to me at the time to pluralize the comment. I am undoing this now.

File: src/cpp/shared/adventure/adv.cpp (hunk 2 above)
Comment (PierceLBrooks) [reply src/cpp/shared/adventure/adv.cpp]: Yes it is. I am simplifying this now. The overly verbose form was just for sanity's sake. Thanks for pointing it out.

**Pull request merged**